lib/apartmentex.ex

```diff
@@ -1,24 +1,18 @@
 defmodule Apartmentex do
-
   @schema_prefix Application.get_env(:apartmentex, :schema_prefix) || "tenant_"
   @tenant_migration_folder "priv/repo/tenant_migrations"
   #warning: don't change this after you already have tenants
 
-
-  alias Ecto.Repo.Queryable
-  alias Ecto.Repo.Model
   alias Ecto.Changeset
-  alias Apartmentex.Helpers.Queryable, as: QHelpers
-  alias Apartmentex.Helpers.Model, as: MHelpers
-  require Ecto.Query
 
   def new_tenant(repo, tenant) do
     prefix = build_prefix(tenant)
     case repo.__adapter__ do
       Ecto.Adapters.Postgres -> Ecto.Adapters.SQL.query(repo, "CREATE SCHEMA #{prefix}", [])
       Ecto.Adapters.MySQL -> Ecto.Adapters.SQL.query(repo, "CREATE DATABASE #{prefix}", [])
     end
-    Apartmentex.Migrator.run(repo, @tenant_migration_folder, :up, [all: true, prefix: String.to_atom(prefix)])
+    Ecto.Migration.Supervisor.start_link
+    Ecto.Migrator.run(repo, @tenant_migration_folder, :up, [all: true, prefix: String.to_atom(prefix)])
```

Comment:
Getting an intermittent test failure, and I wonder if it has to do with the `start_link` here. Is there a better way to ensure that the `Ecto.Migration.Supervisor` is started?

```
  1) test .insert/4 can insert a model (Apartmentex.ApartmentexTest)
     test/apartmentex_test.exs:107
     ** (exit) exited in: GenServer.call(Ecto.Migration.Supervisor, {:start_child, [#PID<0.1580.0>, Apartmentex.TestPostgresRepo, :forward, :up, :info]}, :infinity)
         ** (EXIT) no process
     stacktrace:
       (elixir) lib/gen_server.ex:596: GenServer.call/3
       lib/ecto/migration/runner.ex:22: Ecto.Migration.Runner.run/6
       lib/ecto/migrator.ex:121: Ecto.Migrator.attempt/6
       lib/ecto/migrator.ex:71: anonymous fn/4 in Ecto.Migrator.do_up/4
       lib/ecto/adapters/sql.ex:472: anonymous fn/3 in Ecto.Adapters.SQL.do_transaction/3
       (db_connection) lib/db_connection.ex:973: DBConnection.transaction_run/4
       (db_connection) lib/db_connection.ex:897: DBConnection.run_begin/3
       (db_connection) lib/db_connection.ex:671: DBConnection.transaction/3
       lib/ecto/migrator.ex:244: anonymous fn/4 in Ecto.Migrator.migrate/4
       (elixir) lib/enum.ex:1183: Enum."-map/2-lists^map/1-0-"/2
       test/apartmentex_test.exs:24: Apartmentex.ApartmentexTest.__ex_unit_setup_0/1
       test/apartmentex_test.exs:1: Apartmentex.ApartmentexTest.__ex_unit__/2
```